tests/src/tests.rs

```diff
@@ -414,9 +454,40 @@ fn test(
                 .all(|s| s.starts_with("//") || s.chars().all(|c| c.is_whitespace()));
 
         if is_header {
-            for line in part.lines() {
-                compare_ref = get_flag_metadata(line, "Ref").or(compare_ref);
-                validate_hints = get_flag_metadata(line, "Hints").or(validate_hints);
+            // todo, is this a hack?
+            let source = world.set(&src_path.join("./header"), part.to_string());
```

Comment:
what does this do?

Reply:
It creates a source just for the header (so that it can be parsed) at a path where we don't care so I chose something random, I needed an instance of Source is the whole story

Reply:
I'm using Source::detached now, it's probably okay